source/frontend/src/features/mapSideBar/shared/sidebarPathSolver.ts

```diff
@@ -0,0 +1,152 @@
+import { generatePath, useHistory } from 'react-router-dom';
+
+export const sidebarBasePath = '/mapview/sidebar';
+
+export interface IPathSolverMethods {
+  newFile: (fileType: string) => void;
+  showFile: (fileType: string, fileId: number) => void;
+  showDetail: (fileType: string, fileId: number, detailType: string, replace: boolean) => void;
+  editDetails: (fileType: string, fileId: number, detailType: string) => void;
+  editDetail: (fileType: string, fileId: number, detailType: string, detailId: number) => void;
+  addDetail: (fileType: string, fileId: number, detailType: string) => void;
+  editProperties: (fileType: string, fileId: number) => void;
+  showFilePropertyIndex: (fileType: string, fileId: number, propertyIndex: number) => void;
+  showFilePropertyDetail: (
+    fileType: string,
+    fileId: number,
+    filePropertyId: number,
+    detailType: string,
+    replace: boolean,
+  ) => void;
+}
+
+export type IPathSolver = () => IPathSolverMethods;
```

Comment:
I like this, it is nice to lay out these common *file routing patterns in a shared location.
Perhaps we should avoid the *solver syntax - as so far that has been used to denote a file that calculates authorization.

Reply:
thats fair. Thoughts on a better name?

Reply:
maybe sidebarPathGenerator?